### PR TITLE
[9.0.0] Fix automatic new row/column issues.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This PR should solve problems with automatic row/column creation with HF-driven formulas enabled.

<sup>[skip changelog]</sup>

### How has this been tested?
Added test cases for integrations with the problematic features, like autofill, alter, etc.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
* #8110
* #8121
* #8109
* #8112

### Affected project(s):
- [x] `handsontable`
- [x] `@handsontable/angular`
- [x] `@handsontable/react`
- [x] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Review by aninde — DISMISSED** [review]
All related issues work with this build, well done! 🥇

**Comment by wszymanski** [comment]
One from tests for https://github.com/handsontable/handsontable/issues/8078 will probably need code from this branch 👀

**Review by Siemienik — COMMENTED** [review]
Hi, it looks great for me 👍 

I have only one question about expecting data.

File: src/plugins/formulas/__tests__/featureIntegration.spec.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      await sleep(300);
+
+      expect(hot.getData()).toEqual([
+        ['test', 2, 'TEST', 4, 5, 6],
+        [1, 2, 'TEST', 4, 5, 6],
+        [1, 2, 'TEST', 4, 5, 6],
+        [1, 2, 'TEST', 4, 5, 6],
+        [null, null, 'TEST', null, null, null],
+        [null, null, null, null, null, null]
+      ]);
```
[the remark this answers is not in the document]
Comment (jansiegel) [reply src/plugins/formulas/__tests__/featureIntegration.spec.js:141]: @Siemienik `getData` returns the visual state of the table, what you're referring to is the result of `getSourceData` - if I were to check for source data, I guess it should return exactly that.

[the remark this answers is not in the document]
Comment (Siemienik) [reply src/plugins/formulas/__tests__/featureIntegration.spec.js:141]: Thank you for an explantation, great done

**Review by Siemienik — APPROVED** [review]
🥇

**Pull request merged**